### ISSUE TITLE
No validation error on empty request data

### ISSUE THREAD

**Issue by adeesent** [encode/apistar#286]
Hello, I faced an issue with passing empty request. I expected that if request data is empty, there will be a ValidationError. But instead I get None value in callback function. 

```
class User(Object):
    properties = {
        'name': String
    }

def create_user(user: User):
     pass

routes = [
    Route('/user', 'POST', create_user)
]
```

**Comment by tomchristie** [comment]
Okay. Useful to know: Is there any Content-Type header on the incoming request?

**Comment by adeesent** [comment]
Yeah. Content-Type is empty, and if I set application/json then receive 'Empty JSON' response.

**Comment by pavelholvi** [comment]
Seems easy enough to fix, I'll tackle this one.

**Comment by adeesent** [comment]
Thanks!

**Comment by pavelholvi** [comment]
Not sure if that's a bug or a feature, but either way, created a PR.

**Comment by tomchristie** [comment]
Resolved via #430.
